pkg/controller/mustgather/mustgather_controller.go

```diff
@@ -244,10 +245,28 @@ func (r *ReconcileMustGather) Reconcile(request reconcile.Request) (reconcile.Re
 			if err != nil {
 				reqLogger.Error(err, fmt.Sprintf("Failed to get %s job", instance.Name))
 			} else {
-				err = r.GetClient().Delete(context.TODO(), tmpJob)
-				if err != nil {
-					reqLogger.Error(err, fmt.Sprintf("Failed to delete %s job", instance.Name))
-					return reconcile.Result{}, err
+				// delete pods owned by job
```

Comment:
Interesting. Deleting the Job doesn't do that for you? I guess not, since you added this :P